mmpose/models/heads/heatmap_heads/cid_head.py

```diff
@@ -0,0 +1,727 @@
+        pixel_coords = pixel_coords.float().to(device) + 0.5
+        return pixel_coords
+
+    def forward(self, global_feats: Tensor, instance_feats: Tensor,
+                instance_coords: Tensor) -> Tensor:
+        """Perform spatial attention.
+
+        Args:
+            global_feats (Tensor): Tensor containing the global features.
+            instance_feats (Tensor): Tensor containing the instance feature
+                vectors.
+            instance_coords (Tensor): Tensor containing the root coordinates
+                of the instances.
+
+        Returns:
+            Tensor containing the modulated global features.
+        """
+        B, C, H, W = global_feats.size()
+
+        instance_feats = self.atn(instance_feats).reshape(B, C, 1, 1)
+        feats = global_feats * instance_feats.expand_as(global_feats)
+        fsum = torch.sum(feats, dim=1, keepdim=True)
+
+        pixel_coords = self._get_pixel_coords((W, H), feats.device)
+        relative_coords = instance_coords.reshape(
+            -1, 1, 2) - pixel_coords.reshape(1, -1, 2)
+        relative_coords = relative_coords.permute(0, 2, 1) / 32.0
+        relative_coords = relative_coords.reshape(B, 2, H, W)
+
+        input_feats = torch.cat((fsum, relative_coords), dim=1)
+        mask = self.conv(input_feats).sigmoid()
+        return global_feats * mask
+
+
+class GFDModule(BaseModule):
+    """Global Feature Decoupling module introduced in `CID`. This module
+    extracts the decoupled heatmaps for each instance.
+
+    Args:
+        in_channels (int): Number of channels in the input feature map
+        out_channels (int): Number of channels of the output heatmaps
+            for each instance
+        gfd_channels (int): Number of channels in the transformed feature map
+        clamp_delta (float, optional): A small value that prevents the sigmoid
+            activation from becoming saturated. Defaults to 1e-4.
+        init_cfg (Config, optional): Config to control the initialization. See
+            :attr:`default_init_cfg` for default settings
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        gfd_channels: int,
+        clamp_delta: float = 1e-4,
+        init_cfg: OptConfigType = None,
+    ):
+        super().__init__(init_cfg=init_cfg)
+
+        self.conv_down = build_conv_layer(
+            dict(
+                type='Conv2d',
+                in_channels=in_channels,
+                out_channels=gfd_channels,
+                kernel_size=1))
+
+        self.channel_attention = ChannelAttention(in_channels, gfd_channels)
+        self.spatial_attention = SpatialAttention(in_channels, gfd_channels)
+        self.fuse_attention = build_conv_layer(
+            dict(
+                type='Conv2d',
+                in_channels=gfd_channels * 2,
+                out_channels=gfd_channels,
+                kernel_size=1))
+        self.heatmap_conv = build_conv_layer(
```

Comment:
In mmpose 0.x, heatmap_conv.bias and keypoint_center_conv.bias is initialized as bias_value. Do we still have this feature in 1.x?

bias_value = -math.log((1 - self.prior_prob) / self.prior_prob)
torch.nn.init.constant_(self.keypoint_center_conv.bias, bias_value)
torch.nn.init.constant_(self.heatmap_conv.bias, bias_value)

Reply:
This initialization has not been added yet because the alignment of training accuracy has not been started. Thanks for your reminder